x-pack/plugins/ml/public/application/services/job_service.d.ts

```diff
@@ -21,15 +21,15 @@ declare interface JobService {
   tempJobCloningObjects: {
     createdBy?: string;
     datafeed?: Datafeed;
-    job: any;
+    job?: Job;
     skipTimeRangeStep: boolean;
     start?: number;
     end?: number;
     calendars: Calendar[] | undefined;
   };
   skipTimeRangeStep: boolean;
-  saveNewJob(job: any): Promise<any>;
-  cloneDatafeed(datafeed: any): Datafeed;
+  saveNewJob(job: Job): Promise<any>;
+  cloneDatafeed(Datafeed: Datafeed | undefined): Datafeed;
```

Comment:
this function shouldn't take `undefined` as it doesn't know how to deal with it.
we need a check here to ensure `datafeed` is not `undefined`:
https://github.com/elastic/kibana/blob/3a388c6bf0a9ae115fa1b1be78cd5ab36c669cd4/x-pack/plugins/ml/public/application/jobs/new_job/pages/new_job/page.tsx#L75

Reply:
Updated here https://github.com/elastic/kibana/pull/90078/commits/0fa0b0963d60a2bbc0f74d9cabfe89472b58991a